drv/transceivers-server/src/main.rs

```diff
@@ -620,6 +625,11 @@ fn main() -> ! {
         BUFS.claim()
     };
 
+    #[cfg(feature = "thermal-control")]
+    let thermal_api = Some(Thermal::from(THERMAL.get_task_id()));
+    #[cfg(not(feature = "thermal-control"))]
+    let thermal_api = None;
```

Comment:
this would also change a bit if we follow my above suggestion to remove the `Option`.